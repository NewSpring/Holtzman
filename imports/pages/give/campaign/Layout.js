// @flow
/* eslint-disable react/no-danger */
// $FlowMeteor
import { find } from "ramda";
import { Link } from "react-router";

import Split, { Left, Right } from "../../../components/@primitives/layout/split";
import Meta from "../../../components/shared/meta";
import AddToCart from "../../../components/giving/add-to-cart";

type ILayout = {
  account: Object,
};

const Layout = ({ account }: ILayout) => (
  <div>
    <Split nav classes={["background--light-primary"]}>
      <Meta
        title={account.name}
        description={account.summary}
        image={
          account.images ? find(account.images, { fileLabel: "2:1" }).cloudfront : account.image
        }
        meta={[
          { property: "og:type", content: "article" },
        ]}
      />

      <Right
        background={
          account.images ?
            find(account.images, { fileLabel: "2:1" }).cloudfront :
            account.image
        }
        mobile
      />
      <Right
        background={
          account.images ?
            find(account.images, { fileLabel: "1:2" }).cloudfront :
            account.image
        }
        mobile={false}
      />

    </Split>
    <Left scroll classes={["background--light-primary"]}>
      <Link
        to="/give/now"
        className={
          "locked-top locked-left soft-double@lap-and-up soft " +
          "h7 text-dark-secondary plain visuallyhidden@handheld"
        }
      >
        <i
          className="icon-arrow-back soft-half-right display-inline-block"
          style={{ verticalAlign: "middle" }}
        />
        <span
          className="display-inline-block"
          style={{ verticalAlign: "middle", marginTop: "5px" }}
        >
          Back
        </span>
      </Link>
      <div className="soft@lap-and-up soft-double-top@lap-and-up">
        <div className="soft soft-double-bottom soft-double-top@lap-and-up">

          <h2>{account.name}</h2>
          <div dangerouslySetInnerHTML={{ __html: account.description }} />
        </div>
      </div>

      <div className="background--light-secondary">
        <div className="constrain-copy soft-double@lap-and-up">
          <div className="soft soft-double-bottom soft-double-top@lap-and-up">
            <AddToCart accounts={[account]} donate />
          </div>
        </div>
      </div>

    </Left>
  </div>
);

export default Layout;
