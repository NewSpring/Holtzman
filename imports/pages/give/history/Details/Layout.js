/* eslint-disable react/no-danger */
import { Component, PropTypes } from "react";
import moment from "moment";
import { Link } from "react-router";

import Split, { Left, Right } from "../../../../blocks/split";

import { Spinner } from "../../../../components/loading";
import Currency from "../../../../components/currency";
import SideBySide from "../../../../components/cards/SideBySide";
import Meta from "../../../../components/meta";
import AccountType from "../../../../components/accountType";

export default class Layout extends Component {

  static propTypes = {
    loadingEntries: PropTypes.bool,
    entries: PropTypes.array,
    transaction: PropTypes.object,
  }

  formatDate = (date) => (
    moment(new Date(date)).format("MMM D, YYYY")
  )

  /* eslint-disable max-len */
  render() {
    const { loadingEntries, entries } = this.props;
    return (
      <div>
        <Split nav classes={["background--light-primary"]}>
          <Meta title="Giving History" />
          <Right
            background="//dg0ddngxdz549.cloudfront.net/images/cached/images/remote/http_s3.amazonaws.com/ns.images/all/heroes/newspring/campuses/Florence.1.2x1_1700_850_90_c1.jpg"
            mobile={false}
          />
        </Split>
        <Left scroll ref="container" classes={["background--light-secondary"]}>
          <div
            className={
              "soft-double-sides@lap-and-up soft-double-ends@lap-and-up " +
              "soft background--light-primary"
            }
          >
            {(() => {
              if (process.env.WEB) {
                return (
                  <Link
                    to="/give/history"
                    className={
                      "locked-left soft-double@lap-and-up " +
                      "soft h7 text-dark-secondary plain"
                    }
                  >
                    <i
                      className="icon-arrow-back soft-half-right display-inline-block"
                      style={{ verticalAlign: "middle" }}
                    />
                    <span
                      className="display-inline-block"
                      style={{ verticalAlign: "middle", marginTop: "5px" }}
                    >
                      Back
                    </span>
                  </Link>
                );
              }
              return null;
            })()}
            <div
              className={
                "text-left soft-double-top hard-left@lap-and-up " +
                "soft-half-bottom soft@anchored"
              }
            >
              <div className="soft-double-ends@anchored">
                {(() => {
                  const { transaction } = this.props;
                  if (!transaction) {
                    // loading
                    return (
                      <div className="text-center soft">
                        <Spinner styles={{ width: "40px", height: "40px" }} />
                      </div>
                    );
                  }
                  const { account } = transaction.details[0];
                  const { person } = transaction;
                  return (
                    <div className="text-center">
                      <p className="push-half-bottom">
                        <em>{this.formatDate(transaction.date)}</em>
                      </p>
                      <h3 className="text-dark-secondary">{account.name}</h3>

                      <Currency
                        amount={transaction.details[0].amount.toFixed(2)}
                        baseHeadingSize="1"
                        className="display-inline-block text-center soft-bottom"
                      />

                      <h6 className="push-bottom text-dark-tertiary">
                        {person.nickName || person.firstName} {person.lastName}
                      </h6>

                      {(() => {
                        const detail = transaction.payment;
                        if (detail && detail.accountNumber) {
                          return (
                            <h4 className="text-dark-secondary">
                              {detail.accountNumber.slice(-4)}&nbsp;

                              {(() => {
                                if (detail.paymentType && detail.paymentType === "ACH") {
                                  return (
                                    <AccountType width="30px" height="20px" type="Bank" />
                                  );
                                } else if (detail.paymentType) {
                                  return (
                                    <AccountType
                                      width="30px"
                                      height="20px"
                                      type={detail.paymentType}
                                    />
                                  );
                                }
                                return null;
                              })()}

                            </h4>
                          );
                        }
                        return null;
                      })()}
                      <p
                        className={
                          "text-center soft-ends soft-double@anchored " +
                          "flush-bottom soft-ends soft-sides@portable"
                        }
                      >
                        Thank you so much for your contributions! It is because of your generosity that we are able to continue telling stories of the greatness of Jesus and seeing lives changed.
                      </p>
                    </div>
                  );
                })()}
              </div>
            </div>
          </div>
          <div className="soft-half soft-sides@portable soft-double-sides@anchored">

            <h4 className="soft soft-double-ends text-center@lap-and-up flush-bottom">
              Recent Articles About Giving
            </h4>
            {(() => {
              if (loadingEntries) {
                return <div className="one-whole soft text-center"><Spinner /></div>;
              }
              return (
                <div className="grid">
                  {entries && entries.map((entry, key) => (
                    <div
                      key={key}
                      className={
                        "grid__item one-whole push-half-bottom push-bottom@portable hard-bottom"
                      }
                    >
                      <SideBySide
                        classes={["push-bottom@lap-and-up"]}
                        images={entry.content.images}
                        defaultImage={entry.content.images[0].url}
                      >
                        <h4 className="push-half-top@portable push-top@anchored">
                          {entry.title}
                        </h4>

                        <p><small dangerouslySetInnerHTML={{ __html: entry.meta.summary }} /></p>
                        {(() => {
                          if (process.env.WEB) {
                            return (
                              <a
                                rel="noopener noreferrer"
                                target="_blank"
                                href={`https://newspring.cc/articles/${entry.meta.urlTitle}`}
                                className={
                                  "h6 btn--small btn--dark-tertiary " +
                                  "soft-sides@portable one-whole@handheld"
                                }
                              >
                                Read more
                              </a>
                            );
                          }

                          if (process.env.NATIVE) {
                            return (
                              <Link
                                to={`/articles/${entry.entryId}`}
                                className={
                                  "h6 btn--small btn--dark-tertiary " +
                                  "soft-sides@portable one-whole@handheld"
                                }
                              >
                                Read more
                              </Link>
                            );
                          }

                          return null;
                        })()}


                      </SideBySide>
                    </div>
                  ))}
                </div>
              );
            })()}

          </div>
        </Left>
      </div>

    );
  }
  /* eslint-enable max-len */
}
