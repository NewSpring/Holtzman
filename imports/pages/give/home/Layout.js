// @flow

import Activity from "./Activity";
import Schedules from "./Schedules";
import SavedPayments from "./SavedPayments";

const Layout = () => (
  <div>
    <Activity />
    <Schedules />
    <SavedPayments />
  </div>
);

export default Layout;
