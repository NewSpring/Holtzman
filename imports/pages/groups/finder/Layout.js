/* eslint-disable react/no-danger */
import PropTypes from 'prop-types';

import Meta from "../../../components/shared/meta";
import Loading from "../../../components/@primitives/UI/loading";
import Forms from "../../../components/@primitives/UI/forms";

import GroupFinderFeedItem from "../../../components/content/feed-item-card";
import SideBySide from "../../../components/@primitives/UI/cards/SideBySideCard";

import GroupsILead from "../../../components/groups/groups-i-lead";
import KeywordSelect from "./Fields/Keyword";
import CampusSelect from "./Fields/CampusSelect";
import Location from "./Fields/Location";
import Validate from "../../../util/validate";
import Svg from "../../../components/@primitives/UI/svg";

/* eslint-disable max-len */
const Layout = ({
  tags,
  tagOnClick,
  submitForm,
  canSubmit,
  campuses,
  zip,
  zipDisabled,
  selectedCampus,
  searchQuery,
  inputOnChange,
  content,
  getLocation,
  geoLocationLoading,
}) =>
  (<section className="background--light-secondary hard">
    {/* Meta */}
    <Meta
      title="Group Finder"
      description="Who are your people? We know it's important to be connected, but it's hard to build lasting friendships. What if taking one simple step changed everything? At NewSpring, we’re trying to make it easier for you to find people who share your interests. We know that when you get together with people and have fun, you’ll begin to grow into a strong community that serves and grows together. What if you are one step away from saying, “These are my people”?"
      image="https://dg0ddngxdz549.cloudfront.net/images/cached/images/remote/http_s3.amazonaws.com/ns.images/newspring/groups/groups.2x1_2000_1000_90_a789ae07aae81961.jpg"
    />

    <GroupsILead />

    <div
      className="background--light-primary soft soft-double-top soft-double-top@lap-and-up text-center"
      style={{ overflow: "visible" }}
    >
      <div>
        <h3>Find Your People</h3>
        <h6 className="soft-half-bottom@handheld soft-bottom">
          <em>
            Select your interests, campus, and location <br />to search for
            groups near you.
          </em>
        </h6>
        <Forms.Form
          classes={[
            "soft",
            "one-whole",
            "two-thirds@portable",
            "one-half@anchored",
            "display-inline-block",
            "push-bottom",
          ]}
          submit={submitForm}
          keypress={submitForm}
          action
        >
          <KeywordSelect
            tags={tags}
            searchQuery={searchQuery}
            tagOnClick={tagOnClick}
            onChange={inputOnChange}
          />
          <CampusSelect
            campuses={campuses}
            selectedCampus={selectedCampus}
            onChange={inputOnChange}
          />
          <Location
            zip={zip}
            validation={Validate.isLocationBasedZipCode}
            onChange={inputOnChange}
            disabled={zipDisabled}
            unLocate={getLocation}
          />
          {(() => {
            if (geoLocationLoading) {
              return (
                <div className="one-whole text-center soft">
                  <Loading />
                </div>
              );
            }

            const color =
              !geoLocationLoading && zip === "Using your location"
                ? "#6BAC43"
                : "#505050";

            return (
              <div className={"text-left soft-half-sides"}>
                <a onClick={e => getLocation(e)} style={{ color }}>
                  <Svg
                    name={"locate"}
                    title={"Locate Icon"}
                    fill={color}
                    classes={"display-inline-block"}
                  />
                  <h6
                    className="display-inline-block push-half-left"
                    style={{ fontWeight: "400", verticalAlign: "super" }}
                  >
                    Use my current location
                  </h6>
                </a>
              </div>
            );
          })()}
        </Forms.Form>
      </div>
    </div>
    <div className="soft-double-bottom text-center background--light-primary">
      <div className="background--light-primary soft-double-sides@lap-wide-and-up">
        {(() => {
          if (!tags.length && !campuses.length) {
            return (
              <div className="one-whole text-center">
                <Loading />
              </div>
            );
          } else if (geoLocationLoading) {
            return null;
          }

          const classes = ["btn", "push-top@lap-and-up"];
          if (!canSubmit) {
            classes.push("btn--disabled");
          }
          // XXX why can't I just pass in the function here?
          return (
            <button
              disabled={!canSubmit}
              onClick={submitForm}
              className={classes.join(" ")}
            >
              Let&#39;s Go!
            </button>
          );
        })()}
      </div>
    </div>

    {/* Group Stories */}
    <div
      className={
        "soft-half soft-sides@palm-wide-and-up soft-double-ends " +
        "soft-double@lap-wide-and-up text-center"
      }
    >
      <h3 className="push-top">You Can&#39;t Do Life Alone</h3>
      <div className="grid">
        {content &&
          content.map((entry, key) => {
            if (process.env.WEB) {
              return (
                <a
                  rel="noopener noreferrer"
                  target="_blank"
                  href={`https://newspring.cc/articles/${entry.meta.urlTitle}`}
                  className="plain grid__item one-whole"
                  key={key}
                >
                  <SideBySide
                    classes={["push-bottom@lap-and-up"]}
                    images={entry.content.images}
                    defaultImage={entry.content.images[0].url}
                  >
                    <h4 className="text-dark-primary push-half-top@portable push-top@anchored">
                      {entry.title}
                    </h4>

                    <p className="text-dark-primary">
                      <small
                        dangerouslySetInnerHTML={{ __html: entry.meta.summary }}
                      />
                    </p>
                    <span
                      className={
                        "h6 btn--small btn--dark-tertiary " +
                        "soft-sides@portable one-whole@handheld"
                      }
                    >
                      Read more
                    </span>
                  </SideBySide>
                </a>
              );
            }
            return (
              <div
                className="grid__item one-whole one-half@palm-wide-and-up"
                key={key}
              >
                <GroupFinderFeedItem item={entry} />
              </div>
            );
          })}
      </div>
    </div>
  </section>);
/* eslint-enable max-len */

Layout.propTypes = {
  tags: PropTypes.array.isRequired,
  campuses: PropTypes.array.isRequired,
  zip: PropTypes.string.isRequired,
  selectedCampus: PropTypes.object.isRequired,
  tagOnClick: PropTypes.func.isRequired,
  submitForm: PropTypes.func.isRequired,
  zipDisabled: PropTypes.bool.isRequired,
  canSubmit: PropTypes.bool.isRequired,
  searchQuery: PropTypes.array.isRequired,
  inputOnChange: PropTypes.func.isRequired,
  content: PropTypes.array.isRequired,
  getLocation: PropTypes.func.isRequired,
  geoLocationLoading: PropTypes.bool.isRequired,
};

export default Layout;
