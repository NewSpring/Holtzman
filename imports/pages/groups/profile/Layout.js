import { PropTypes } from "react";
import { Link } from "react-router";
import { Meteor } from "meteor/meteor";

import Meta from "../../../components/shared/meta";

const rockUrl = Meteor.settings.public.rock.baseURL;

const Layout = ({ group, leaders, isLeader, join, loginParam }) => (
  <section className="background--light-secondary hard">
    {/* Meta */}
    <Meta
      title={group.name}
      image={group.photo}
      description={group.description}
      id={group.id}
    />

    {/* Hero */}
    <div
      className="ratio--landscape background--fill"
      style={{
        overflow: "visible",
        backgroundImage: `url('${group.photo}')`,
        zIndex: 10,
      }}
    >
      <div className="ratio__item" />
    </div>

    {/* Group Information */}
    <div
      style={{ borderRadius: "0 0 6px 6px" }}
      className={
        "push-double-sides@lap-wide-and-up push-half-sides " +
        "flush-ends card outlined outlined--light"
      }
    >
      <div className="card__item soft">
        <h3 className="text-dark-primary push-half-top push-bottom">{group.name}</h3>

        <h7 className="text-dark-secondary">
          <small>Group Leaders</small>
        </h7>
        <h5 className="text-dark-secondary soft-half-top">
          {leaders.map((x, i) => {
            let string = `${x.person.nickName || x.person.firstName} ${x.person.lastName}`;
            if (leaders.length - 1 !== i) string += ", ";
            return <span key={i}>{string}</span>;
          })}
        </h5>
        {leaders.map((leader, i) => (
          <div
            className="ratio--square round display-inline-block push-right background--fill"
            key={i}
            style={{
              backgroundImage: `url('${leader.person.photo}')`,
              width: "80px",
              height: "80px",
            }}
          >
            <div className="ratio__item" />
          </div>
        ))}
      </div>
    </div>

    {/* Main card stacks */}
    <section
      className="soft-double-sides@lap-wide-and-up soft-half-sides soft-half-ends flush-sides"
    >

      {/* Join Group CTA */}
      {/* shows manage group if you are a leader / can manage */}
      <div className="card outlined outlined--light soft-sides-@lap-and-up">
        <div className="grid card__item soft push-half flush">
          <h4
            className={
              "flush-bottom hard-left push-bottom@handheld push-bottom@lap " +
              "push-half-top grid__item one-half@lap-wide-and-up one-whole " +
              "text-center@handheld text-center@lap text-dark-primary"
            }
          >
            #TheseAreMyPeople
          </h4>
          <div
            className={
              "grid__item hard-left text-right@lap-wide-and-up " +
              "text-center one-whole one-half@lap-wide-and-up"
            }
          >
            {(() => {
              const className = "flush-bottom push-half-bottom@handheld btn";
              if (isLeader) {
                return (
                  <a
                    rel="noopener noreferrer"
                    target="_blank"
                    className={className}
                    href={
                      `${rockUrl}page/521?GroupId=${group.entityId}&${loginParam}`
                    }
                  >
                    Manage Group
                  </a>
                );
              }

              let newloginParam = loginParam;

              if (loginParam) {
                newloginParam = `&${loginParam}`;
              }

              return (
                <a
                  rel="noopener noreferrer"
                  target="_blank"
                  className={className}
                  href={
                    `${rockUrl}Workflows/304?Group=${group.guid}${newloginParam}`
                  }
                >
                  Contact
                </a>
              );
            })()}
          </div>
        </div>
      </div>

      {/* General Information */}
      <div className="card outlined outlined--light hard one-whole">
        <div className="card__item push-half-top@handheld">
          <div className="soft-left@lap-wide-and-up soft soft-double-bottom">

            <h5 className="soft-half-ends">Group Details</h5>

            {/* Group Meeting Schedule */}
            {(() => {
              if (!group.schedule) return null;
              if (!group.schedule.description) return null;
              return (
                <div className="soft-bottom">
                  <h7 className="text-dark-secondary">Time</h7>
                  <h6 className="text-dark-secondary soft-half-top flush-bottom">
                    {group.schedule.description}
                  </h6>
                </div>
              );
            })()}

            {/* Group Location */}
            {(() => {
              if (!group.locations) return null;
              const loc = group.locations[0];
              if (!loc) return null;
              return (
                <div className="soft-bottom">
                  <h7 className="text-dark-secondary">
                    Address
                  </h7>
                  <h6 className="text-dark-secondary soft-half-top flush-bottom">
                    {loc.location.city}, {loc.location.state}
                  </h6>
                </div>
              );
            })()}

            {/* Campus  */}
            {(() => {
              if (!group.campus || !group.campus.name) return null;
              return (
                <div className="soft-bottom">
                  <h7 className="text-dark-secondary">
                    Campus
                  </h7>
                  <h6 className="text-dark-secondary soft-half-top flush-bottom">
                    {group.campus.name}
                  </h6>
                </div>
              );
            })()}

            {/* Group Information */}
            <div>
              <h7 className="text-dark-secondary">Information</h7>
              <h6 className="text-dark-secondary soft-half-top flush-bottom">
                {group.kidFriendly ? "Children Welcome" : "Adults Only"}
                {group.ageRange ? `, ${group.ageRange[0]} - ${group.ageRange[1]}` : ""}
              </h6>
            </div>

          </div>
        </div>
      </div>

      {/* More Information */}
      <div className="card outlined outlined--light hard one-whole">
        <div className="card__item push-half-top@handheld">
          <div className="soft-left@lap-wide-and-up soft soft-bottom">

            <h5 className="soft-half-ends">More Information</h5>

            {/* Group Description */}
            <div className="soft-double-bottom@lap-wide-and-up soft-bottom">
              <h7 className="text-dark-secondary">Description</h7>
              <p className="soft-half-top flush-bottom">
                {group.description && group.description.split("\n").map((text, key) => (
                  <span key={key}>
                    {text}
                    <br />
                  </span>
                ))}
              </p>
            </div>

            {/* Group Members */}
            <div className="soft-double-bottom@lap-wide-and-up soft-bottom">
              <h7 className="text-dark-secondary">Members</h7>
              <div className="soft-half-top flush-bottom">
                {group.members.filter((x) => x.person && x.person.photo).map((member, i) => (
                  <div
                    className={
                      "ratio--square round display-inline-block " +
                      "push-half-right background--fill"
                    }
                    key={i}
                    style={{
                      backgroundImage: `url('${member.person.photo}')`,
                      width: "40px",
                      height: "40px",
                    }}
                  >
                    <div className="ratio__item" />
                  </div>
                ))}
              </div>
            </div>

            {/* Tags */}
            <div>
              <h7 className="text-dark-secondary">Tags</h7>
              <div className="soft-half-top flush-bottom">
                {group.tags && group.tags.map((tag, i) => (
                  <span className="tag push-half-right" key={i}>{tag.value}</span>
                ))}
                {(() => {
                  if (!group.type || group.type === "Interests") return null;
                  return <span className="tag push-half-right">{group.type}</span>;
                })()}
                {(() => {
                  if (!group.kidFriendly) return null;
                  return <span className="tag push-half-right">kid friendly</span>;
                })()}
                {(() => {
                  if (!group.demographic) return null;
                  return <span className="tag push-half-right">{group.demographic}</span>;
                })()}
              </div>
            </div>
          </div>
        </div>
      </div>

      {/* Search */}
      <div className="card outlined outlined--light soft-sides-@lap-and-up push-bottom">
        <div className="card__item soft-double">
          <h4 className=" push-bottom one-whole text-center text-dark-secondary">
            Looking for another group?
          </h4>
          <div className="text-center one-whole">
            <Link
              to="/groups/finder"
              className="flush-bottom push-half-bottom@handheld btn--dark-tertiary"
            >
              Find A Group
            </Link>
          </div>
        </div>
      </div>
    </section>
  </section>
);

Layout.propTypes = {
  group: PropTypes.object.isRequired,
  leaders: PropTypes.array.isRequired,
  isLeader: PropTypes.bool.isRequired,
  join: PropTypes.func.isRequired,
};

export default Layout;
