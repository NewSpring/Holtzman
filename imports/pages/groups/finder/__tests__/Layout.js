import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import Layout from "../Layout";

const defaultProps = {
  tags: [
    { value: "test" },
    { value: "test" },
  ],
  tagOnClick: jest.fn(),
  submitTags: jest.fn(),
  canSearchTags: true,
  findByQuery: jest.fn(),
  inputOnChange: jest.fn(),
  content: [{}, {}],
};

const generateComponent = (additionalProps = {}) => {
  const newProps = {
    ...defaultProps,
    ...additionalProps,
  };
  return <Layout { ...newProps } />;
};

it("renders with props", () => {
  const wrapper = shallow(generateComponent());
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it("renders disabled tags", () => {
  const wrapper = shallow(generateComponent({
    canSearchTags: false,
  }));
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it("renders loading if no tags", () => {
  const wrapper = shallow(generateComponent({
    tags: [],
  }));
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});
