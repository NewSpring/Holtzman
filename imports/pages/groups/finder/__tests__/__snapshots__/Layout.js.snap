// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders disabled tags 1`] = `
<section
  className="background--light-secondary hard"
>
  <Meta
    description="Who are your people? We know it's important to be connected, but it's hard to build lasting friendships. What if taking one simple step changed everything? At NewSpring, we’re trying to make it easier for you to find people who share your interests. We know that when you get together with people and have fun, you’ll begin to grow into a strong community that serves and grows together. What if you are one step away from saying, “These are my people”?"
    image="https://dg0ddngxdz549.cloudfront.net/images/cached/images/remote/http_s3.amazonaws.com/ns.images/newspring/groups/groups.2x1_2000_1000_90_a789ae07aae81961.jpg"
    title="Group Finder"
  />
  <Apollo(Component) />
  <div
    className="background--light-primary soft soft-double-top soft-double-top@lap-and-up text-center"
    style={
      Object {
        "overflow": "visible",
      }
    }
  >
    <div>
      <h3>
        Find Your People
      </h3>
      <h6
        className="soft-half-bottom@handheld soft-bottom"
      >
        <em>
          Select your interests, campus, and location 
          <br />
          to search for groups near you.
        </em>
      </h6>
      <Form
        action={true}
        classes={
          Array [
            "soft",
            "one-whole",
            "two-thirds@portable",
            "one-half@anchored",
            "display-inline-block",
            "push-bottom",
          ]
        }
      >
        <Keywords
          onChange={[Function]}
          tagOnClick={[Function]}
          tags={
            Array [
              Object {
                "value": "test",
              },
              Object {
                "value": "test",
              },
            ]
          }
        />
        <CampusSelect
          onChange={[Function]}
        />
        <Location
          onChange={[Function]}
          validation={[Function]}
        />
        <div
          className="text-left soft-half-sides"
        >
          <a
            onClick={[Function]}
            style={
              Object {
                "color": "#505050",
              }
            }
          >
            <Svg
              classes="display-inline-block"
              fill="#505050"
              height="24px"
              name="locate"
              title="Locate Icon"
              width="24px"
            />
            <h6
              className="display-inline-block push-half-left"
              style={
                Object {
                  "fontWeight": "400",
                  "verticalAlign": "super",
                }
              }
            >
              Use my current location
            </h6>
          </a>
        </div>
      </Form>
    </div>
  </div>
  <div
    className="soft-double-bottom text-center background--light-primary"
  >
    <div
      className="background--light-primary soft-double-sides@lap-wide-and-up"
    >
      <button
        className="btn push-top@lap-and-up btn--disabled"
        disabled={true}
      >
        Let's Go!
      </button>
    </div>
  </div>
  <div
    className="soft-half soft-sides@palm-wide-and-up soft-double-ends soft-double@lap-wide-and-up text-center"
  >
    <h3
      className="push-top"
    >
      You Can't Do Life Alone
    </h3>
    <div
      className="grid"
    >
      <div
        className="grid__item one-whole one-half@palm-wide-and-up"
        key="0"
      >
        <Connect(Apollo(LikesWrapper))
          item={Object {}}
        />
      </div>
      <div
        className="grid__item one-whole one-half@palm-wide-and-up"
        key="1"
      >
        <Connect(Apollo(LikesWrapper))
          item={Object {}}
        />
      </div>
    </div>
  </div>
</section>
`;

exports[`renders loading if no tags 1`] = `
<section
  className="background--light-secondary hard"
>
  <Meta
    description="Who are your people? We know it's important to be connected, but it's hard to build lasting friendships. What if taking one simple step changed everything? At NewSpring, we’re trying to make it easier for you to find people who share your interests. We know that when you get together with people and have fun, you’ll begin to grow into a strong community that serves and grows together. What if you are one step away from saying, “These are my people”?"
    image="https://dg0ddngxdz549.cloudfront.net/images/cached/images/remote/http_s3.amazonaws.com/ns.images/newspring/groups/groups.2x1_2000_1000_90_a789ae07aae81961.jpg"
    title="Group Finder"
  />
  <Apollo(Component) />
  <div
    className="background--light-primary soft soft-double-top soft-double-top@lap-and-up text-center"
    style={
      Object {
        "overflow": "visible",
      }
    }
  >
    <div>
      <h3>
        Find Your People
      </h3>
      <h6
        className="soft-half-bottom@handheld soft-bottom"
      >
        <em>
          Select your interests, campus, and location 
          <br />
          to search for groups near you.
        </em>
      </h6>
      <Form
        action={true}
        classes={
          Array [
            "soft",
            "one-whole",
            "two-thirds@portable",
            "one-half@anchored",
            "display-inline-block",
            "push-bottom",
          ]
        }
      >
        <Keywords
          onChange={[Function]}
          tagOnClick={[Function]}
          tags={Array []}
        />
        <CampusSelect
          campuses={Array []}
          onChange={[Function]}
        />
        <Location
          onChange={[Function]}
          validation={[Function]}
        />
        <div
          className="text-left soft-half-sides"
        >
          <a
            onClick={[Function]}
            style={
              Object {
                "color": "#505050",
              }
            }
          >
            <Svg
              classes="display-inline-block"
              fill="#505050"
              height="24px"
              name="locate"
              title="Locate Icon"
              width="24px"
            />
            <h6
              className="display-inline-block push-half-left"
              style={
                Object {
                  "fontWeight": "400",
                  "verticalAlign": "super",
                }
              }
            >
              Use my current location
            </h6>
          </a>
        </div>
      </Form>
    </div>
  </div>
  <div
    className="soft-double-bottom text-center background--light-primary"
  >
    <div
      className="background--light-primary soft-double-sides@lap-wide-and-up"
    >
      <div
        className="one-whole text-center"
      >
        <Loading />
      </div>
    </div>
  </div>
  <div
    className="soft-half soft-sides@palm-wide-and-up soft-double-ends soft-double@lap-wide-and-up text-center"
  >
    <h3
      className="push-top"
    >
      You Can't Do Life Alone
    </h3>
    <div
      className="grid"
    >
      <div
        className="grid__item one-whole one-half@palm-wide-and-up"
        key="0"
      >
        <Connect(Apollo(LikesWrapper))
          item={Object {}}
        />
      </div>
      <div
        className="grid__item one-whole one-half@palm-wide-and-up"
        key="1"
      >
        <Connect(Apollo(LikesWrapper))
          item={Object {}}
        />
      </div>
    </div>
  </div>
</section>
`;

exports[`renders with props 1`] = `
<section
  className="background--light-secondary hard"
>
  <Meta
    description="Who are your people? We know it's important to be connected, but it's hard to build lasting friendships. What if taking one simple step changed everything? At NewSpring, we’re trying to make it easier for you to find people who share your interests. We know that when you get together with people and have fun, you’ll begin to grow into a strong community that serves and grows together. What if you are one step away from saying, “These are my people”?"
    image="https://dg0ddngxdz549.cloudfront.net/images/cached/images/remote/http_s3.amazonaws.com/ns.images/newspring/groups/groups.2x1_2000_1000_90_a789ae07aae81961.jpg"
    title="Group Finder"
  />
  <Apollo(Component) />
  <div
    className="background--light-primary soft soft-double-top soft-double-top@lap-and-up text-center"
    style={
      Object {
        "overflow": "visible",
      }
    }
  >
    <div>
      <h3>
        Find Your People
      </h3>
      <h6
        className="soft-half-bottom@handheld soft-bottom"
      >
        <em>
          Select your interests, campus, and location 
          <br />
          to search for groups near you.
        </em>
      </h6>
      <Form
        action={true}
        classes={
          Array [
            "soft",
            "one-whole",
            "two-thirds@portable",
            "one-half@anchored",
            "display-inline-block",
            "push-bottom",
          ]
        }
      >
        <Keywords
          onChange={[Function]}
          tagOnClick={[Function]}
          tags={
            Array [
              Object {
                "value": "test",
              },
              Object {
                "value": "test",
              },
            ]
          }
        />
        <CampusSelect
          onChange={[Function]}
        />
        <Location
          onChange={[Function]}
          validation={[Function]}
        />
        <div
          className="text-left soft-half-sides"
        >
          <a
            onClick={[Function]}
            style={
              Object {
                "color": "#505050",
              }
            }
          >
            <Svg
              classes="display-inline-block"
              fill="#505050"
              height="24px"
              name="locate"
              title="Locate Icon"
              width="24px"
            />
            <h6
              className="display-inline-block push-half-left"
              style={
                Object {
                  "fontWeight": "400",
                  "verticalAlign": "super",
                }
              }
            >
              Use my current location
            </h6>
          </a>
        </div>
      </Form>
    </div>
  </div>
  <div
    className="soft-double-bottom text-center background--light-primary"
  >
    <div
      className="background--light-primary soft-double-sides@lap-wide-and-up"
    >
      <button
        className="btn push-top@lap-and-up btn--disabled"
        disabled={true}
      >
        Let's Go!
      </button>
    </div>
  </div>
  <div
    className="soft-half soft-sides@palm-wide-and-up soft-double-ends soft-double@lap-wide-and-up text-center"
  >
    <h3
      className="push-top"
    >
      You Can't Do Life Alone
    </h3>
    <div
      className="grid"
    >
      <div
        className="grid__item one-whole one-half@palm-wide-and-up"
        key="0"
      >
        <Connect(Apollo(LikesWrapper))
          item={Object {}}
        />
      </div>
      <div
        className="grid__item one-whole one-half@palm-wide-and-up"
        key="1"
      >
        <Connect(Apollo(LikesWrapper))
          item={Object {}}
        />
      </div>
    </div>
  </div>
</section>
`;
